Index: easyshop/shop/service.py
IDEA additional info:
Subsystem: com.intellij.openapi.diff.impl.patch.BaseRevisionTextPatchEP
<+>from django.shortcuts import get_object_or_404\r\nfrom .models import Product, Category\r\nfrom typing import List\r\n\r\nfrom shop.models import Category\r\nclass ProductService:\r\n    def get_all_products(self):\r\n        fields = ['id', 'slug', 'name', 'price', 'image', 'category']\r\n        return Product.products.only(*fields)\r\n\r\n    def get_product_by_slug(self, slug):\r\n        fields = ['id', 'slug', 'name', 'price', 'image', 'category']\r\n        product = get_object_or_404(Product.products.only(*fields), slug=slug)\r\n        return product\r\n\r\n    def get_products_by_category_slug(self, category_slug):\r\n\r\n        return Product.products.filter(category__slug=category_slug)\r\n\r\n\r\nclass CategoryService:\r\n    @staticmethod\r\n    def get_categories() -> List[Category]:\r\n        return Category.objects.all()\r\n\r\n    @staticmethod\r\n    def get_category_by_slug(slug: str) -> Category:\r\n        return get_object_or_404(Category, slug=slug)
===================================================================
diff --git a/easyshop/shop/service.py b/easyshop/shop/service.py
--- a/easyshop/shop/service.py	
+++ b/easyshop/shop/service.py	
@@ -3,6 +3,7 @@
 from typing import List
 
 from shop.models import Category
+
 class ProductService:
     def get_all_products(self):
         fields = ['id', 'slug', 'name', 'price', 'image', 'category']
